include/zephyr/net/wifi.h

```diff
@@ -50,14 +50,16 @@ enum wifi_security_type {
 	WIFI_SECURITY_TYPE_SAE,
 	/** GB 15629.11-2003 WAPI security. */
 	WIFI_SECURITY_TYPE_WAPI,
-	/** EAP security - Enterprise. */
-	WIFI_SECURITY_TYPE_EAP,
 	/** WEP security. */
 	WIFI_SECURITY_TYPE_WEP,
 	/** WPA-PSK security. */
 	WIFI_SECURITY_TYPE_WPA_PSK,
 	/** WPA/WPA2/WPA3 PSK security. */
 	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL,
+	/** EAP security - Enterprise. */
+	WIFI_SECURITY_TYPE_EAP,
```

Comment:
this would break shell as the enumerations change, so, better leave it in the original place.

Reply:
actually in the feature, we will add more enterprise mode wifi_security_type, like:
	/** EAP PEAP MSCHAPV2 security - Enterprise. */
	WIFI_SECURITY_TYPE_EAP_PEAP_MSCHAPV2,
	/** EAP PEAP GTC security - Enterprise. */
	WIFI_SECURITY_TYPE_EAP_PEAP_GTC,
	/** EAP TTLS MSCHAPV2 security - Enterprise. */
	WIFI_SECURITY_TYPE_EAP_TTLS_MSCHAPV2,
	/** EAP PEAP TLS security - Enterprise. */
	WIFI_SECURITY_TYPE_EAP_PEAP_TLS,
	/** FT-PSK security */
	WIFI_SECURITY_TYPE_FT_PSK,
	/** FT-SAE security */
	WIFI_SECURITY_TYPE_FT_SAE,
	/** FT-EAP security */
	WIFI_SECURITY_TYPE_FT_EAP,
	/** FT-EAP-SHA384 security */
	WIFI_SECURITY_TYPE_FT_EAP_SHA384,
and the shell cmd like:
		  "[-k, --key-mgmt]: Key Management type (valid only for secure SSIDs)\n"
		  "0:None, 1:WPA2-PSK, 2:WPA2-PSK-256, 3:SAE, 4:WAPI, 5:WEP, 6: WPA-PSK\n"
		  "7: WPA-Auto-Personal, 8: DPP, 9: EAP-TLS, 10: EAP-PEAP-MSCHAPv2\n"
		  "11: EAP-PEAP-GTC, 12: EAP-TTLS-MSCHAPv2, 13: EAP-PEAP-TLS\n"

	so it's weird to keep the old one WIFI_SECURITY_TYPE_EAP. I suggest to directly remove WIFI_SECURITY_TYPE_EAP.
	if must keep it to keep the backwards, it should move to place with other enterprise security type.

Reply:
> actually in the feature, we will upstream more enterprise mode wifi_security_type(already support in our downstream), 

Sure, but we will remove EAP eventually, though this is just an Enumeration, still better to keep is compatible (See https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated for guidelines regards to API).

We can mark EAP as deprecated and recommend EAP_TLS as the replacement and in the future can remove EAP.

Reply:
> this would break shell as the enumerations change
could you share which case the shell will be broken? As the shell cmd has prompt in console, user can check the cmd usage, and the EAP type is not widely used currently in zephyr.
> though this is just an Enumeration, still better to keep is compatible (See https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated for guidelines regards to API
From this page, I only see that mark "function declarations or macro definitions" as deprecated

Reply:
> > this would break shell as the enumerations change
> 
> could you share which case the shell will be broken? As the shell cmd has prompt in console, user can check the cmd usage, and the EAP type is not widely used currently in zephyr.

The numbers for the security types are changed, not just EAP, but for WEP and WPA-PSK too.

> > though this is just an Enumeration, still better to keep is compatible (See https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated for guidelines regards to API
> 
> From this page, I only see that mark "function declarations or macro definitions" as deprecated. How to mark the enumeration as deprecated?

As I said, no clear instructions for Enumeration, but if we don't need to break it, it's best both old and new co-exist, at least for a while.

Reply:
udpated

Reply:
Thank you.